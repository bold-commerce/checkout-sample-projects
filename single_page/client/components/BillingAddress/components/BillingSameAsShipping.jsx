import { RadioField } from '@boldcommerce/stacks-ui';
import React from 'react';
import { useTranslation } from 'react-i18next';

const BillingSameAsShipping = ({
  billingSameAsShipping,
  setBillingSameAsShipping,
  disabled,
}) => {
  const { t } = useTranslation();
  return (
  <>
    <RadioField
      label={t('billing.same_as_shipping')}
      name="billing-address"
      className="RadioButton"
      value="SAME_AS_SHIPPING_ADDRESS"
      checked={billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(true)}
      disabled={disabled}
    />
    <RadioField
      label={t('billing.different_from_shipping')}
      name="billing-address"
      className="RadioButton"
      value="DIFFERENT_BILLING_ADDRESS"
      checked={!billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(false)}
      disabled={disabled}
    />
  </>
  );
};

export default BillingSameAsShipping;
