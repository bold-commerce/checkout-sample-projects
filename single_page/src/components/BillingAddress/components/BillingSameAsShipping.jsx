import { RadioField } from '@boldcommerce/stacks-ui';
import React from 'react';

const BillingSameAsShipping = ({
  billingSameAsShipping,
  setBillingSameAsShipping,
  disabled,
}) => (
  <>
    <RadioField
      label="Same as shipping address"
      name="billing-address"
      className="RadioButton"
      value="SAME_AS_SHIPPING_ADDRESS"
      checked={billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(true)}
      disabled={disabled}
    />
    <RadioField
      label="Use a different billing address"
      name="billing-address"
      className="RadioButton"
      value="DIFFERENT_BILLING_ADDRESS"
      checked={!billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(false)}
      disabled={disabled}
    />
  </>
);

export default BillingSameAsShipping;
