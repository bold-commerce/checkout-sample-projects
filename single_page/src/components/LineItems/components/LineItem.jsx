import React from 'react';
import PropTypes from 'prop-types';
import Product from './Product';
import { Price } from '@boldcommerce/stacks-ui';
import { useTranslation } from 'react-i18next';

const LineItem = ({
  title,
  image,
  quantity,
  totalPrice,
  variants
}) => {
  const { t } = useTranslation();
  return(
    <div className="SummaryBlock CartItem" >
      <Product
        title={title}
        image={image}
        variants={variants}
      />
      <div className="CartItem__QuantityPriceWrapper">
        <div className="CartItem__ProductQuantityWrapper">
          <div className="ProductQuantity" aria-label="product quantity">
            {quantity}
          </div>
        </div>
        <div className="CartItem__ProductPrice">
          <Price amount={totalPrice} moneyFormatString={t('currency_format')} />
        </div>
      </div>
    </div>
  );}
  
  LineItem.propTypes = {
    title: PropTypes.string,
    image: PropTypes.string,
    quantity: PropTypes.number,
    lineItemKey: PropTypes.string,
    variants: PropTypes.array,
  };
  
  export default LineItem;