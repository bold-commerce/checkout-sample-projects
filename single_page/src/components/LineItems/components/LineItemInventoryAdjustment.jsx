import React from 'react';
import { LineItemQuantity } from '.';
import ArrowRightIcon from './ArrowRightIcon';
import { useTranslation } from 'react-i18next';

const LineItemInventoryAdjustment = ({ originalQuantity, quantity, readOnly }) => {
  const { t } = useTranslation();
  if (quantity === 0) {
    return (
      <div className="LineItem__QuantityAdjustmentWrapper">
        <span className="LineItem__SoldOut">{t('line_item.sold_out')}</span>
      </div>
    );
  } else {
    return (
      <div className="LineItem__QuantityAdjustmentWrapper">
        <LineItemQuantity
          readOnly={readOnly}
          defaultValue={originalQuantity}
        />
        <ArrowRightIcon />
        <LineItemQuantity
          readOnly={readOnly}
          defaultValue={quantity}
        />
      </div>
    )
  }
};

export default LineItemInventoryAdjustment;
