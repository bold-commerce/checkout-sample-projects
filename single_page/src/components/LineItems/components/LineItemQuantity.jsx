import React, { useCallback, useState } from 'react';
import classNames from 'classnames';
import { Button, Input } from '@boldcommerce/stacks-ui';

const LineItemQuantity = ({
  defaultValue,
  onChange,
  readOnly,
}) => {
  const [value, setValue] = useState(defaultValue);

  const increment = useCallback(() => {
    setValue(value + 1);
    onChange(value + 1);
  }, [value, setValue, onChange]);

  const decrement = useCallback(() => {
    setValue(value - 1);
    onChange(value - 1);
  }, [value, setValue, onChange]);

  const handleChange = useCallback((e) => {
    setValue(e.target.value);
  }, [setValue]);

  const handleBlur = useCallback((e) => {
    const [val] = e.target.value.split('.');
    const num = parseInt(val, 10);

    // Reseting the quantity to previous state if the value entered is invalid
    if (Number.isNaN(num) || num < 1) {
      setValue(defaultValue);
      return;
    }

    setValue(num);
    onChange(num);
  }, [value, onChange, setValue, defaultValue]);

  return (
    <div className="LineItem__LineItemQuantityWrapper">
      <div className="LineItemQuantity__Decrement">
        {!readOnly && (
          <Button disabled={typeof value === 'string' || value <= 1} onClick={decrement}>-</Button>
        )}
      </div>
      <div className={classNames('LineItemQuantity', { 'LineItemQuantity--read-only': readOnly })}>
        {
          readOnly ? <span className="LineItemQuantity__ReadOnlyValue">{value}</span> : (
            <Input
              readOnly={readOnly}
              type="number"
              value={value}
              onChange={handleChange}
              onBlur={handleBlur}
            />
          )
        }
      </div>
      <div className="LineItemQuantity__Increment">
        {!readOnly && (
          <Button disabled={typeof value === 'string'} onClick={increment}>+</Button>
        )}
      </div>
    </div>
  );
};

export default LineItemQuantity;
