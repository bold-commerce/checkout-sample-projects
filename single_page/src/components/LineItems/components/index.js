export { default as LineItem } from './LineItem';
export { default as Product } from './Product';
