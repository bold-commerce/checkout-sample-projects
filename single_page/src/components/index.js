export * from './Address';
export * from './BillingAddress';
export * from './CheckoutButton';
export * from './CheckoutSection';
export * from './Customer';
export * from './DiscountForm';
export * from './EmptyState';
export * from './Header';
export * from './LineItems';
export * from './LoadingState';
export * from './OrderErrors';
export * from './OrderSummary';
export * from './PaymentMethod';
export * from './ShippingAddress';
export * from './ShippingLines';
