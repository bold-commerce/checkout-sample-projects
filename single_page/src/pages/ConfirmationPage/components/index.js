export * from './ConfirmationList';
export * from './ConfirmationListItem';
