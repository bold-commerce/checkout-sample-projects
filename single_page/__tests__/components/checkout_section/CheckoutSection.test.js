import React from "react";
import { render } from "@testing-library/react";
import { CheckoutSection } from '../../../client/components/CheckoutSection';
import { exampleLineItems as MOCKlineItems } from '../../utils/lineItemHelpers';
import {
  countries as MOCKcountries,
  caProvinces as MOCKprovinces,
  exampleAddress as MOCKaddress
} from '../../utils/addressHelpers';

jest.mock('@boldcommerce/checkout-react-components', () => ({
  useBillingAddress: () => ({
    data: {MOCKaddress},
    submitBillingAddress: (() => {})
  }),
  useBillingSameAsShipping: () => ({
    setBillingSameAsShipping: (() => {}),
    data: true
  }),
  useCountryInfo: () => ({
    data: {
      countries: MOCKcountries,
      provinces: MOCKprovinces,
      provinceLabel: "province",
      showPostalCode: true,
      showProvince: true
    }
  }),
  usePaymentIframe : () => ({
    processPaymentIframe: (() => {})
  }),
  useCheckoutStore: () => ({
    state: {
      orderInfo: 'fulfilled',
      loadingStatus: {
        isLoading: false
      }
    }
  }),
  useLineItems: () => ({ 
    data : MOCKlineItems
  })
}))

describe('CheckoutSection', () => {
  test('renders CheckoutSection component without children', () => {
    const { asFragment } = render(
      <CheckoutSection 
        title="Test Title"
      />
    )

    expect(asFragment()).toMatchSnapshot();
  })

  test('renders CheckoutSection component with children', () => {
    const { asFragment } = render(
      <CheckoutSection 
        title="Test Title"
        children={
          <>
            <div>test child one</div>
            <div>test child two</div>
            <div>test child three</div>
          </>
        }
      />
    )

    expect(asFragment()).toMatchSnapshot();
  })
})
