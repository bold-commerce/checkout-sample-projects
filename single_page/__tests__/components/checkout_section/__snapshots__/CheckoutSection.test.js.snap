// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutSection renders CheckoutSection component with children 1`] = `
<DocumentFragment>
  <section
    class="FieldSet undefined"
  >
    <div
      class="FieldSet__Header"
    >
      <h3
        class="FieldSet__Heading"
      >
        Test Title
      </h3>
    </div>
    <div
      class="FieldSet__Content"
    >
      <div>
        test child one
      </div>
      <div>
        test child two
      </div>
      <div>
        test child three
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`CheckoutSection renders CheckoutSection component without children 1`] = `
<DocumentFragment>
  <section
    class="FieldSet undefined"
  >
    <div
      class="FieldSet__Header"
    >
      <h3
        class="FieldSet__Heading"
      >
        Test Title
      </h3>
    </div>
    <div
      class="FieldSet__Content"
    />
  </section>
</DocumentFragment>
`;
